File: monitoring_activite/objects/activite.php
<?php

class Activite{
	
	private $_id_activite;
	private $_date_debut;
	private $_date_fin;
	private $_type_activite;

	//Constructeur
	public function __construct(array $activite){
		
		$this->_id_activite = (int) $activite['id_activite'];
		$this->_date_debut = $activite['date_debut'];
		$this->_date_fin = $activite['date_fin'];
		$this->_type_activite = $activite['type_activite'];
	}

	//Getters
	public function id_activite() { return $this->_id_activite; }
	public function date_debut() { return $this->_date_debut; }
	public function date_fin() { return $this->_date_fin; }
	public function type_activite() { return $this->_type_activite; }
	
}
?>

File: Programmes-Qt/WSListener/activite.cpp
#include "activite.h"

Activite::Activite(QDateTime date_debut, QDateTime date_fin, QString type_activite, unsigned int id_activite, bool debug):
    m_id_activite(id_activite),
    m_date_debut(date_debut),
    m_date_fin(date_fin),
    m_type_activite(type_activite),
    m_debug(debug){

}

Activite::Activite(){

}

unsigned int Activite::get_id_activite() const{
    return m_id_activite;
}

QDateTime Activite::get_date_debut() const{
    return m_date_debut;
}

QDateTime Activite::get_date_fin() const{
    return m_date_fin;
}

QString Activite::get_type_activite() const{
    return m_type_activite;
}
